src/gt4py/next/embedded/nd_array_field.py

```diff
@@ -203,15 +203,12 @@ def remap(
 
     __call__ = remap  # type: ignore[assignment]
 
-    def restrict(self, index: common.AnyIndexSpec) -> common.Field | core_defs.ScalarT:
+    def restrict(self, index: common.AnyIndexSpec) -> NdArrayField:
         new_domain, buffer_slice = self._slice(index)
-
         new_buffer = self.ndarray[buffer_slice]
-        if len(new_domain) == 0:
-            # TODO: assert core_defs.is_scalar_type(new_buffer), new_buffer
-            return new_buffer  # type: ignore[return-value] # I don't think we can express that we return `ScalarT` here
-        else:
-            return self.__class__.from_array(new_buffer, domain=new_domain)
+        if new_domain.ndim == 0:
+            return self.from_array(new_buffer, domain=new_domain)
+        return self.__class__.from_array(new_buffer, domain=new_domain)
```

Comment:
I think the `if` is redundant, aren't the two branches doing the same thing?

Reply:
Damn you're right